src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -82,12 +82,12 @@ public class SubmissionController {
             halt(400, "Invalid repo url");
             return null;
         }
-        SubmissionDao submissionDao = DaoService.getSubmissionDao();
-        Submission submission = submissionDao.getSubmissionsForPhase(user.netId(), request.getPhase()).stream()
-                .filter(s -> s.headHash().equals(headHash))
-                .findFirst()
-                .orElse(null);
-        if (submission != null) {
+        if (mostRecentHasMaxScore(netId, request.getPhase())) {
+            halt(400,"You have already earned the highest possible score on this phase");
+            throw new RuntimeException("Max score already earned on this phase");
```

Comment:
Sorry, I know I initially said to throw a runtime exception, but I think this would should instead be `return null;` to match the other guards in this class

Reply:
consider it done, because its done